.github/workflows/ci-publish.yml

```diff
@@ -85,17 +87,22 @@ jobs:
         with:
           python-version: '3.9'
 
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Should this step need a working copy of the source code, and git histories? The publish step should use the built wheel, so I don't think it should need the source code or the tags (the fetch-depth parameter).

I had rather liked the simplicity of the previous step that did not checkout the code; it simply grabbed the wheel artefact and published to PyPI with twine. Would that still be appropriate here?

Reply:
To explain why this step is necessary: the Poetry Publish step looks at the version in pyproject.toml and finds files related to the version being published, then publishes only those files. If it can't find files related to the current version it doesn't publish anything. The problem was that Poetry was unable to get the 'current' version without the source code for the versioning plugin to calculate it.

With regards to the overall question of whether we should revert to Twine, I see the arguments as follows: we now have a working system that keeps the dependencies, runtime, build system and publish all within one ecosystem, which I consider to be a preferable in the long term. If we do revert back to Twine, that would have to be re-tested, however it would require less steps in the publish action, which would make these steps easier to follow and maintain.

I don't have strong feelings either way, and will revert to Twine if that is considered the better option.

Reply:
Makes sense, thanks for the explanation. I agree with your points about keeping it in the same ecosystem. Would suggest adding a comment to this step just summarising the above, as I think it's a bit of a subtle point